src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -320,6 +328,18 @@ private void OnPost()
             } while (wasWork && loopsRemaining > 0);
         }
 
+        private void OnHeartbeat(UvTimerHandle timer)
+        {
+            Walk(ptr =>
+            {
+                var now = _engine.Libuv.hrtime();
+                //var now = DateTime.UtcNow.Ticks;
```

Comment:
Might also mean you can use its interface (that's created for testing it), to jump time around and test the timeouts more easily - without actually timing out.